array-manipulation-3/src/intersection.js

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable no-unused-vars */
+
+function intersection(first, second) {
+  const joinedArray = first.concat(second);
+  let resultArray = [];
+
+  for (let i = 0; i < first.length; i++) {
+    for (let x = 0; x < second.length; x++) {
+      for (let y = 0; y < joinedArray.length; y++) {
+        if (first[i] === joinedArray[y] && second[x] === joinedArray[y]) {
+          resultArray.push(first[i]);
+          resultArray = unique(resultArray);
+        }
+      }
+    }
+  }
```

Comment:
Again this is a _lot_ of looping. Avoid putting everything in one array at the beginning, the approach is similar to my comment on `difference`.